sources/sql_database/schema_types.py

```diff
@@ -139,13 +139,15 @@ def table_to_columns(
     table: Table,
     reflection_level: ReflectionLevel = "full",
     type_conversion_fallback: Optional[TTypeAdapter] = None,
+    included_columns: Optional[List[str]] = None,
 ) -> TTableSchemaColumns:
     """Convert an sqlalchemy table to a dlt table schema."""
     return {
         col["name"]: col
         for col in (
             sqla_col_to_column_schema(c, reflection_level, type_conversion_fallback)
             for c in table.columns
+            if included_columns is None or c.name in included_columns
```

Comment:
this certainly works. maybe we could remove columns from the Table object? so for example when table adapter is called user sees right set of columns?

or rename included_columns to selected_columns :)

Reply:
yep deleting the sqla columns is simpler. Doing this now in the default table adapter https://github.com/dlt-hub/verified-sources/pull/550/files#diff-5d691408095c848d7a6e68ba31e82c21c396fb280c655ad5f09d68f579bba94eR41